PR_REVIEW_CHECKLIST.md

```diff
@@ -0,0 +1,89 @@
+# PR review checklist
+
+This document lists specific checks that should be done when reviewing a PR. This means these items should be checked by:
```

Comment:
```suggestion
This document lists specific guidelines for all PR reviews and aims to clarify items in the PR checklist.
```
Since this document is not a checklist I think this change would make more sense. If these guidelines apply to all reviewers then we should just say that and remove the numbered list below.

Reply:
Nice, since marc gave similar feedback I'll remove the itemized list below :)